provider_shopper/test/login_widget_test.dart

```diff
@@ -0,0 +1,44 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:provider/provider.dart';
+import 'package:provider_shopper/models/cart.dart';
+import 'package:provider_shopper/models/catalog.dart';
+import 'package:provider_shopper/screens/catalog.dart';
+import 'package:provider_shopper/screens/login.dart';
+
+void main() {
+  testWidgets('Login page Widget test', (tester) async {
+    await tester.pumpWidget(MultiProvider(
+      providers: [
+        Provider(create: (context) => CatalogModel()),
+        ChangeNotifierProxyProvider<CatalogModel, CartModel>(
+          create: (context) => CartModel(),
+          update: (context, catalog, cart) {
+            cart.catalog = catalog;
+            return cart;
+          },
+        ),
+      ],
+      child: MaterialApp(
+        initialRoute: '/',
+        routes: {
+          '/': (context) => MyLogin(),
+          '/catalog': (context) => MyCatalog(),
+        },
+      ),
+    ));
+
+    // Verifying the behaviour of ENTER button.
+    final enterBtnFinder = find.widgetWithText(RaisedButton, 'ENTER');
```

Comment:
These two lines can be removed. If there's no widget with 'ENTER' as its text, the finder on line 39 will fail, and there's no way there will ever be two or more. 😄